aequilibrae/paths/linear_approximation.py

```diff
@@ -331,10 +358,19 @@ def derivative_of_objective(stepsize):
             return np.sum(self.congested_value * (self.step_direction_flow - self.fw_total_flow))
 
         try:
-            min_res = root_scalar(derivative_of_objective, bracket=[0, 1])
-            self.stepsize = min_res.root
-            if not min_res.converged:
-                logger.warn("Descent direction stepsize finder is not converged")
+            if recent_scipy:
+                min_res = root_scalar(derivative_of_objective, bracket=[0, 1])
+                self.stepsize = min_res.root
+                if not min_res.converged:
+                    logger.warn("Descent direction stepsize finder is not converged")
+            else:
+                min_res = root_scalar(derivative_of_objective, 1 / self.iter)
+                if not min_res.success:
+                    logger.warn("Descent direction stepsize finder is not converged")
+                self.stepsize = min_res.x[0]
+                if self.stepsize <= 0.0 or self.stepsize >= 1.0:
```

Comment:
this should be <0 and >1, ==0 and ==1 are handled as edge cases below

Reply:
This is the only way I got things to work with this older version of SciPy.  Results we get from them are a bit weird (and quite a bit slower too)

Reply:
wait, it's wrapped in a try block so it'll do the right thing when it catches the exception, sorry I missed that bit of logic, all good